### PR TITLE
Give more information on why the broker was not found 

### PR DESCRIPTION
**Pull request by aikar**
for BrokerNotAvailable

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -303,10 +303,15 @@ Client.prototype.sendGroupRequest = function (encode, decode, requestArgs) {
 
   var request = encode.apply(null, requestArgs);
   var broker = this.brokerForLeader(this.coordinatorId);
-
-  if (!broker || !broker.socket || broker.socket.error || broker.socket.destroyed) {
+  var brokerError = null;
+  if (!broker || !broker.socket) {
```
Comment (hyperlink) [inline lib/client.js]: Do we need this check since the socket should always exist on a broker if it's available?

File: lib/client.js [hunk 2]
```diff
@@ -303,10 +303,15 @@ Client.prototype.sendGroupRequest = function (encode, decode, requestArgs) {
...
+    brokerError = 'Could not find broker';
+  } else if (broker.socket.error || broker.socket.closing || broker.socket.destroyed) {
```
Comment (hyperlink) [inline lib/client.js]: we can use `broker.isConnected`

**Comment by aikar** [comment]
@hyperlink changes nadd

**Comment by hyperlink** [comment]
Published as 3.0.0

**Pull request merged**